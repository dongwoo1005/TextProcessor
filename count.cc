#include "count.h"
#include <sstream>
using namespace std;

// Refer to count.h for implementation details

void Count::setSource(istream *in){
	tp.setSource(in);
}

string Count::to_string(int n){
	ostringstream ss;
	ss << n;
	return ss.str();
}

string Count::getWord(){
	string s = tp.getWord();
	int len = s.length();
	for (int i=0; i<len; ++i){
		if (s[i] == c){
			count++;
			s = s.substr(0,i) + to_string(count) + s.substr(i+1);
		}
	}
	return s;
}

bool Count::fail() const {
	return tp.fail();
}
