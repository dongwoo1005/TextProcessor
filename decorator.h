#ifndef __DECORATOR_H__
#define __DECORATOR_H__
#include "textprocess.h"

class Decorator: public TextProcessor{
protected:
	TextProcessor& tp;
	Decorator(TextProcessor& tp): tp(tp){}
	~Decorator(){ delete &tp;}
};

#endif
