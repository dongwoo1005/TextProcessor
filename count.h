#ifndef __COUNT_H__
#define __COUNT_H__
#include "decorator.h"
#include <string>

// Concrete Decorator Count

class Count: public Decorator {
	char c;		// character c to be counted
	int count;	// current count of the charcter c
	// string to_string(int n)
	// takes integer n as a input to convert it into the string form.
	std::string to_string(int n);	

public:
	Count(TextProcessor& tp, char c): Decorator(tp), c(c), count(0) {}  // ctor
	
	// void setSource(istream *inp)
	// sets the istream source as inp
	void setSource (std::istream* inp); 

	// string getWord()
	// returns the string in the istream source
	// The occurence of the character c in the string is replaced with count number
	std::string getWord();

	// bool fail() const
	// returns the failed variable in the class field
	bool fail() const;
};

#endif
