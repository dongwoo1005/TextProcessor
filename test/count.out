one1
two2
three3
four4
five5
six6
seven7
eight8
nine9
ten10
